solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -210,6 +217,12 @@ public void addCloseHook(Closeable hook) {
     }
   }
 
+  public QueryLimits getLimits() {
+    return req == null
+        ? QueryLimits.NONE
+        : (QueryLimits) req.getContext().computeIfAbsent(LIMITS_KEY, (k) -> new QueryLimits(req));
```

Comment:
I think this is still incorrect... Since it uses the latest SRI from the stack it will completely ignore the already existing QueryLimits created at the bottom of the stack, and create a new instance to track just this part of the processing - whereas we want to track the resource consumption across the whole end-to-end request processing, beginning with the first SRI (at the bottom of the stack) allocated to this query.